### PR TITLE
Fix incompatibility between cupy.random.permutation and numpy.random.permutation .

### LINKED ISSUE #1137: cupy.random.permutation() overwrites its argument.
`cupy.random.permutation()` overwrites its argument.
This is incompatible with `numpy.random.permutation`.

### PR DESCRIPTION
**Pull request by asi1024**
Fixed #1137.

### CODE REVIEW

File: tests/cupy_tests/random_tests/test_permutations.py [hunk 1]
```diff
@@ -12,6 +12,41 @@ class TestPermutations(unittest.TestCase):
...
+        a = testing.shaped_random((), xp)
+        xp.random.permutation(a)
+
+    # Test same values
+
+    @testing.for_all_dtypes(no_float16=True, no_bool=True, no_complex=True)
+    def test_permutation_sort_1dim(self, dtype):
+        a = cupy.arange(10, dtype=dtype)
+        b = cupy.copy(a)
+        cupy.random.permutation(a)
```
Comment (okuta) [inline tests/cupy_tests/random_tests/test_permutations.py]: Why do not use returned value?

Comment (asi1024) [reply tests/cupy_tests/random_tests/test_permutations.py]: Fixed.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**